### PR TITLE
Catch unhandled errors on both frontend and backend

### LINKED ISSUE #3284: [REF-2825] Need a mechanism to catch unhandled errors from event handlers
**Describe the bug**
Currently reflex catches all unhandled exceptions and throws an rx.window_alert; this behavior is not customizable 😔

Related to [https://github.com/orgs/reflex-dev/discussions/2509](https://github.com/orgs/reflex-dev/discussions/2509)

<sub>From [SyncLinear.com](https://synclinear.com) | [REF-2825](https://linear.app/reflex-dev/issue/REF-2825/need-a-mechanism-to-catch-unhandled-errors-from-event-handlers)</sub>

### PR DESCRIPTION
**Pull request by maximvlah**
continuation of #3476

closes #3284

### CODE REVIEW

**Comment by maximvlah** [comment]
@masenf here it is, please test it out and let me know, looking forward to hearing your thoughts on [this](https://github.com/reflex-dev/reflex/pull/3476#issuecomment-2181170365)

**Comment by benedikt-bartscher** [comment]
@maximvlah reflex-web ci is failing

**Comment by Lendemor** [comment]
Will review/test it this week.

File: reflex/components/base/error_boundary.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+from reflex.vars import Var
+
+
+class ErrorBoundary(Component):
+    """A React Error Boundary component that catches unhandled frontend exceptions."""
+
+    library = "react-error-boundary"
+    tag = "ErrorBoundary"
+
+    def _render(self, props: Union[Dict[str, Any], None] = None) -> Tag:
```
Comment (masenf) [inline reflex/components/base/error_boundary.py]: it still works if we just defined the given props with defaults rather that overriding `_render` with almost the same logic as the parent. will push a commit here

File: reflex/components/component.py [hunk 2]
```diff
@@ -1350,6 +1350,9 @@ def _get_imports(self) -> ParsedImportDict:
         # Get static imports required for event processing.
         event_imports = Imports.EVENTS if self.event_triggers else {}
 
+        # Get static imports required for error handling.
+        errors_imports = Imports.FRONTEND_ERRORS if self.tag == "ErrorBoundary" else {}
```
Comment (masenf) [inline reflex/components/component.py]: these imports should just be added to the ErrorBoundary component instead of special casing them here

**Review by masenf — COMMENTED** [review]
pushed some updates

File: reflex/components/base/error_boundary.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        Custom code is inserted at module level, after any imports.
+
+        Returns:
+            The custom code to add.
+        """
+        fallback_container = div(
+            p("Ooops...Unknown Reflex error has occured:"),
+            p(
+                Var.create("error.message", _var_is_local=False, _var_is_string=False),
+                color_scheme="red",
```
[the remark this answers is not in the document]
Comment (benedikt-bartscher) [reply reflex/components/base/error_boundary.py:60]: @maximvlah could you please address this?

[the remark this answers is not in the document]
Comment (maximvlah) [reply reflex/components/base/error_boundary.py]: @benedikt-bartscher done

**Review by Lendemor — APPROVED** [review]

**Review by benedikt-bartscher — APPROVED** [review]

**Review by masenf — APPROVED** [review]

**Pull request merged**